src/core/resolver/xds/xds_dependency_manager.cc

```diff
@@ -629,6 +629,28 @@ void XdsDependencyManager::PopulateDnsUpdate(const std::string& dns_name,
   dns_state->update.endpoints = std::move(resource);
 }
 
+std::string XdsDependencyManager::GenerateResolutionNoteForCluster(
+    absl::string_view cluster_resolution_note,
+    absl::string_view endpoint_resolution_note) const {
+  std::array<absl::string_view, 4> notes = {
+      lds_resolution_note_, rds_resolution_note_, cluster_resolution_note,
+      endpoint_resolution_note};
+  std::vector<absl::string_view> resolution_notes;
+  for (const auto& note : notes) {
+    if (!note.empty()) resolution_notes.push_back(note);
+  }
+  std::string node_id_buffer;
+  if (resolution_notes.empty()) {
+    const XdsBootstrap::Node* node =
+        DownCast<const GrpcXdsBootstrap&>(xds_client_->bootstrap()).node();
+    if (node != nullptr) {
+      node_id_buffer = absl::StrCat("xDS node ID:", node->id());
```

Comment:
nit, I believe this makes it easier to read
```suggestion
      node_id_buffer = absl::StrCat("xDS node ID: ", node->id());
```

Reply:
That's a pre-existing format, not something new in this PR, and it would require changing a whole bunch of tests to change it.  Feel free to send me a separate PR to do that if you'd like.